src/app/dataset-create/dataset-create.service.ts

```diff
@@ -100,34 +112,88 @@ export class AppDatasetCreateService {
-                            .subscribe();
-                        this.navigationService.navigateToDatasetView({
-                            accountName,
-                            datasetName,
-                            tab: DatasetViewTypeEnum.Overview,
-                        });
-                    }
+        return this.getIdByAccountNameAndDatasetName(
+            accountName,
+            datasetName,
+        ).pipe(
+            switchMap((id: string) =>
+                this.datasetApi.commitEvent({
+                    datasetId: id,
+                    event,
                 }),
-            );
+            ),
+            map((data: CommitEventToDatasetMutation | undefined | null) => {
+                if (
+                    data?.datasets.byId?.metadata.chain.commitEvent
+                        .__typename === "CommitResultAppendError" ||
+                    data?.datasets.byId?.metadata.chain.commitEvent
+                        .__typename === "MetadataManifestMalformed"
+                ) {
+                    this.errorCommitEventChanges(
+                        data.datasets.byId.metadata.chain.commitEvent.message,
+                    );
+                } else {
+                    this.successActions(accountName, datasetName);
+                }
+            }),
+        );
+    }
+
+    public getIdByAccountNameAndDatasetName(
+        accountName: string,
+        datasetName: string,
+    ): Observable<string> {
+        const key = `${accountName}${datasetName}`;
+        if (this.cache.has(key)) {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            return of(this.cache.get(key)!);
+        } else {
+            return this.datasetApi
+                .getDatasetInfoByAccountAndDatasetName(accountName, datasetName)
+                .pipe(
+                    map((data: DatasetByAccountAndDatasetNameQuery) => {
+                        const id = data.datasets.byOwnerAndName?.id as string;
+                        this.cache.set(key, id);
+                        return id;
+                    }),
+                );
+        }
+    }
+
+    public updateReadme(
+        accountName: string,
+        datasetName: string,
+        content: string,
+    ): Observable<void> {
+        return this.getIdByAccountNameAndDatasetName(
+            accountName,
+            datasetName,
+        ).pipe(
+            switchMap((id: string) =>
+                this.datasetApi.updateReadme(id, content),
+            ),
+            map((data: UpdateReadmeMutation | null | undefined) => {
+                if (
+                    data?.datasets.byId?.metadata.updateReadme.__typename ===
+                    "CommitResultSuccess"
+                ) {
+                    this.successActions(accountName, datasetName);
+                }
+            }),
+        );
+    }
+
+    private successActions(accountName: string, datasetName: string): void {
```

Comment:
commit success actions?
Does not matter, if we split the interfaces

Reply:
Fixed